__tests__/components/Button/Button.spec.tsx

```diff
@@ -0,0 +1,275 @@
+    });
+
+    const button = document.querySelector('a');
+
+    button.click();
+    expect(button).not.toBeNull();
+    expect(fn).toBeCalledTimes(1);
+  });
+
+  it('should render a small button', () => {
+    act(() => {
+      ReactDOM.render(<Button size="small" />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      height: 30px;
+      padding: 0px 20px;
+      font-size: 12px;
+    `);
+  });
+
+  it('should render a big button', () => {
+    act(() => {
+      ReactDOM.render(<Button size="big" />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      height: 40px;
+      padding: 0px 30px;
+      font-size: 16px;
+    `);
+  });
+
+  it('should render a default circle button', () => {
+    act(() => {
+      ReactDOM.render(<Button shape="circle" />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      width: 36px;
+      height: 36px;
+      padding: 0px;
+      font-size: 14px;
+      line-height: 41px;
+      border-radius: 50%;
+    `);
+  });
+
+  it('should render a small circle button', () => {
+    act(() => {
+      ReactDOM.render(<Button shape="circle" size="small" />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      width: 30px;
+      height: 30px;
+      padding: 0px;
+      font-size: 12px;
+      line-height: 35px;
+      border-radius: 50%;
+    `);
+  });
+
+  it('should render a big circle button', () => {
+    act(() => {
+      ReactDOM.render(<Button shape="circle" size="big" />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      width: 40px;
+      height: 40px;
+      padding: 0px;
+      font-size: 16px;
+      line-height: 45px;
+      border-radius: 50%;
+    `);
+  });
+
+  it('should render a block button', () => {
+    act(() => {
+      ReactDOM.render(<Button displayBlock />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      width: 100%;
+    `);
+  });
+
+  it('should render a flat button', () => {
+    act(() => {
+      ReactDOM.render(<Button flat />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      color: rgb(85, 85, 85);
+      box-shadow: none;
+      padding: 0px 5px;
+      background: transparent;
+    `);
+  });
+
+  it('should render a outline button', () => {
+    act(() => {
+      ReactDOM.render(<Button outline />, container);
+    });
+
+    const button = document.querySelector('button');
+
+    expect(button).toHaveStyle(`
+      color: rgb(85, 85, 85);
+      box-shadow: none;
+      border: 1px solid #555555;
+      background: transparent;
+    `);
+  });
+
+  it('should not onClick when loading', () => {
```

Comment:
```suggestion
  it('should not allow click while loading', () => {
```

Reply:
Resolved in 2dc849680c3d8db7d149e9eddfc74994ee0b93c8